### DISCUSSION TITLE
Add a setting to allow only values available in options for dashboard templating

### DISCUSSION THREAD

**Discussion by jiangpengcheng** [grafana/grafana#40510] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

What I want is to verify option get from url and if it doesn't exist in available options, reject that value, an example described here: https://github.com/grafana/grafana/pull/35493#issuecomment-861181212

**Why is this needed**:

I think this should be useful to limit user trying non-exist option on some sensible dashboard variables

My user case for this feature is: there is one dashboard with a `service` variable and opened for all users in my Grafana, I want to make user only able to view his/her own services' metrics, so I used a json datasource for `service` variable and only return related `services` options according to user info, this works fine, but one flaw is that user can change `service` variable to any value via url and make him/her access `services` not belong to him/her.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
